scripts/debugging_scripts/check_eth_status.py

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+# SPDX-FileCopyrightText: (c) 2025 Tenstorrent AI ULC
+#
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Usage:
+    scripts/debugging_scripts/check_eth_status.py
+
+Description:
+    Check status on the ethernet cores
+"""
+
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from ttexalens.context import Context
+from ttexalens.device import Device, OnChipCoordinate
+from ttexalens.register_store import read_word_from_device
+
+from check_per_device import run as get_check_per_device
+
+from ttexalens.hw.tensix.blackhole.blackhole import BlackholeDevice
+from ttexalens.hw.tensix.wormhole.wormhole import WormholeDevice
+from scripts.debugging_scripts import utils
+from triage import ScriptConfig, log_check, run_script
+
+script_config = ScriptConfig(
+    depends=["check_per_device"],
+)
+
+
+@dataclass
+class EthCoreDefinitions:
+    port_status: int
+    retrain_count: int
+    rx_link_up: int
+    heartbeat: int
+    mailbox: int
+    mailbox_slots: int
+
+
+class EthCore(ABC):
+    """
+    Base class for Ethernet cores that provides common functionality.
+    """
+
+    def __init__(self, location: OnChipCoordinate, context: Context):
+        self.location = location
+        self.context = context
+        self.eth_core_definitions = EthCoreDefinitions(**self.get_addresses())
+
+    @abstractmethod
+    def get_addresses(self) -> dict:
+        """Return architecture-specific addresses as a dictionary."""
+        pass
+
+    @abstractmethod
+    def port_status_to_string(self, port_status: int) -> str:
+        """Convert port status value to a readable string."""
+        pass
+
+    def check_for_heartbeat(self) -> tuple[bool, str]:
+        """Check for heartbeat at the heartbeat address."""
+        previous_data = 0
+        # Check for a changing value at the heartbeat address. Read up to 100 times
+        for i in range(100):
+            read_data = read_word_from_device(self.location, self.eth_core_definitions.heartbeat, context=self.context)
+            if read_data != previous_data:
+                return True, ""
+            previous_data = read_data
+        return False, f"No heartbeat detected for {self.location.to_user_str()}"
```

Comment:
```suggestion
    def check_for_heartbeat(self):
        """Check for heartbeat at the heartbeat address."""
        previous_data = 0
        # Check for a changing value at the heartbeat address. Read up to 100 times
        for i in range(100):
            read_data = read_word_from_device(self.location, self.eth_core_definitions.heartbeat, context=self.context)
            if read_data != previous_data:
                return
            previous_data = read_data
        log_check(False, f"No heartbeat detected for {self.location.to_user_str()}")
```

After that, you don't need `log_check` on line 77